src/main/java/uk/gov/hmcts/reform/unspec/service/FeesService.java

```diff
@@ -40,15 +34,13 @@ private FeeLookupResponseDto lookupFee(ClaimValue claimValue) {
         );
     }
 
-    private BigInteger getFeeAmountInPence(FeeLookupResponseDto feeLookupResponseDto) {
-        var feeAmountPounds = feeLookupResponseDto.getFeeAmount();
-
-        return feeAmountPounds.multiply(PENCE_PER_POUND).toBigInteger();
-    }
+    private Fee buildFeeDto(FeeLookupResponseDto feeLookupResponseDto) {
+        BigDecimal calculatedAmount = feeLookupResponseDto.getFeeAmount()
+            .multiply(PENCE_PER_POUND)
+            .setScale(0, RoundingMode.CEILING);
```

Comment:
I think we should always use `RoundingMode.UNNECESSARY`. When there's going to be some weird value it's better to throw exception than charge more (even if it's only one pence).

Reply:
sorry I think you mentioned this too me before, will update.